gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenColumnExpressionSuite.scala

```diff
@@ -16,4 +16,40 @@
  */
 package org.apache.spark.sql
 
-class GlutenColumnExpressionSuite extends ColumnExpressionSuite with GlutenSQLTestsTrait {}
+import org.apache.spark.sql.execution.ProjectExec
+import org.apache.spark.sql.functions.{expr, input_file_name}
+import org.apache.spark.sql.types._
+
+class GlutenColumnExpressionSuite extends ColumnExpressionSuite with GlutenSQLTestsTrait {
+  testGluten("input_file_name with scan is fallback") {
+    withTempPath {
+      dir =>
+        val rawData = Seq(
+          Row(1, "Alice", Seq(Row(Seq(1, 2, 3)))),
+          Row(2, "Bob", Seq(Row(Seq(4, 5)))),
+          Row(3, "Charlie", Seq(Row(Seq(6, 7, 8, 9))))
+        )
+        val schema = StructType(
+          Array(
+            StructField("id", IntegerType, nullable = false),
+            StructField("name", StringType, nullable = false),
+            StructField(
```

Comment:
The test data seems complicated, do we need this complicated test data to test `input_file_name`?

Reply:
yes, we need this to cover scenario for scan fallback due to complex data type with project has `input_file_name`